cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1515,14 +1530,34 @@ public void getAvailableTopics(ChannelTopicsListener listener) {
         return topics;
     }
 
+    public void addSubscriptionTags(String[] tagIds) {
+        currentPositionOfTagToAdd = 0;
+        addSubscriptionTag(tagIds[currentPositionOfTagToAdd], new OnAddTagCompleted() {
+            @Override
+            public void onAddTagCompleted() {
+                if (currentPositionOfTagToAdd != tagIds.length - 1) {
+                    currentPositionOfTagToAdd++;
+                    addSubscriptionTag(tagIds[currentPositionOfTagToAdd], this::onAddTagCompleted);
+                }
+            }
+        });
+    }
+
     public void addSubscriptionTag(String tagId) {
-        Log.d("CleverPush", "addSubscriptionTag: " + tagId);
+        addSubscriptionTag(tagId, null);
+    }
+
+    public void addSubscriptionTag(String tagId, OnAddTagCompleted onAddTagCompleted) {
         this.waitForTrackingConsent(() -> new Thread(() -> this.getSubscriptionId(subscriptionId -> {
             if (subscriptionId != null) {
                 Set<String> tags = this.getSubscriptionTags();
                 if (tags.contains(tagId)) {
+                    if (onAddTagCompleted != null) {
+                        onAddTagCompleted.onAddTagCompleted();
```

Comment:
please rename either the method or the interface, as this does not look good to read